### PR TITLE
Bump up iotdb-session to 0.12.4 and fix IoTDB storage option bug

### PR DESCRIPTION
**Pull request by LIU-WEI-git**
1. Fix IoTDB Storage Option insert null index value.
2. Set the default value of SW_STORAGE_IOTDB_SESSIONPOOL_SIZE to 8. If it's 0, the SessionPool size will be 2 * CPU_Cores
3. Bump up iotdb-session to 0.12.4

- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -45,6 +45,9 @@ Release Notes.
 * Fix `SelfRemoteClient` self observing metrics.
 * Add env variables `SW_CLUSTER_INTERNAL_COM_HOST` and `SW_CLUSTER_INTERNAL_COM_PORT` for cluster selectors `zookeeper`,`consul`,`etcd` and `nacos`.
 * Doc update: `configuration-vocabulary`,`backend-cluster` about env variables `SW_CLUSTER_INTERNAL_COM_HOST` and `SW_CLUSTER_INTERNAL_COM_PORT`.
+* Fix IoTDB Storage Option insert null index value.
+* Set the default value of SW_STORAGE_IOTDB_SESSIONPOOL_SIZE to 2 * CPU_Cores.
```
Comment (wu-sheng) [inline CHANGES.md:49]: Should be less?  Check PersistenceTimer codes, the default prepare thread is only 2.

```java
prepareExecutorService = Executors.newFixedThreadPool(moduleConfig.getPrepareThreads());
```

How many threads are you building in the bulk metric and segment processing?

Comment (LIU-WEI-git) [reply CHANGES.md]: The default value of SW_STORAGE_IOTDB_SESSIONPOOL_SIZE is 8, while 2 for low-resource e2e.

File: docs/en/setup/backend/backend-storage.md [hunk 2]
```diff
@@ -294,7 +294,7 @@ storage:
     username: ${SW_STORAGE_IOTDB_USERNAME:root}
     password: ${SW_STORAGE_IOTDB_PASSWORD:root}
     storageGroup: ${SW_STORAGE_IOTDB_STORAGE_GROUP:root.skywalking}
-    sessionPoolSize: ${SW_STORAGE_IOTDB_SESSIONPOOL_SIZE:0} # Zero means the size is 2*CPU_Cores
+    sessionPoolSize: ${SW_STORAGE_IOTDB_SESSIONPOOL_SIZE:8} # Zero means the size is 2*CPU_Cores
```
Comment (wu-sheng) [inline docs/en/setup/backend/backend-storage.md:297]: Please adjust the comments to follow the codes.

Comment (LIU-WEI-git) [reply docs/en/setup/backend/backend-storage.md]: OK

File: oap-server/server-starter/src/main/resources/application.yml [hunk 3]
```diff
@@ -258,7 +258,7 @@ storage:
     username: ${SW_STORAGE_IOTDB_USERNAME:root}
     password: ${SW_STORAGE_IOTDB_PASSWORD:root}
     storageGroup: ${SW_STORAGE_IOTDB_STORAGE_GROUP:root.skywalking}
-    sessionPoolSize: ${SW_STORAGE_IOTDB_SESSIONPOOL_SIZE:16}
+    sessionPoolSize: ${SW_STORAGE_IOTDB_SESSIONPOOL_SIZE:8} # If it's zero, the SessionPool size will be 2*CPU_Cores
```
Comment (wu-sheng) [inline oap-server/server-starter/src/main/resources/application.yml:261]: `configuration-vocabulary.md` should be updated accordingly.

Comment (LIU-WEI-git) [reply oap-server/server-starter/src/main/resources/application.yml:261]: Got it.

**Review by wu-sheng — APPROVED** [review]

**Comment by wu-sheng** [comment]
Let's see whether IoTDB cases could be more stable.

**Pull request merged**